packages/@react-spectrum/dialog/stories/AlertDialog.stories.tsx

```diff
@@ -17,175 +17,158 @@ import {singleParagraph} from './Dialog.stories';
-  children: singleParagraph(),
-  primaryActionLabel: 'Accept',
-  cancelLabel: 'Cancel',
-  secondaryActionLabel: 'Secondary button',
-  onPrimaryAction: action('primary'),
-  onSecondaryAction: action('secondary'),
-  onCancel: action('cancel'),
-  autoFocusButton: 'primary'
-});
-
-AutoFocusPrimary.story = {
+export const AutoFocusPrimary = {
+  render: renderAlert,
+  args: {
+    ...Error.args,
+    secondaryActionLabel: 'Secondary button',
+    autoFocusButton: 'primary'
+  },
   name: 'autoFocus primary'
 };
 
-export const SecondaryDisabled = () => renderAlert({
-  variant: 'error',
-  title: 'Error: Danger Will Robinson',
-  children: singleParagraph(),
-  primaryActionLabel: 'Accept',
-  secondaryActionLabel: 'Secondary button',
-  cancelLabel: 'Cancel',
-  onPrimaryAction: action('primary'),
-  onSecondaryAction: action('secondary'),
-  onCancel: action('cancel'),
-  isSecondaryActionDisabled: true
-});
-
-SecondaryDisabled.story = {
+export const SecondaryDisabled = {
+  render: renderAlert,
+  args: {
+    ...Error.args,
+    secondaryActionLabel: 'Secondary button',
+    isSecondaryActionDisabled: true
+  },
   name: 'secondary disabled'
 };
 
-export const AutoFocusSecondary = () => renderAlert({
-  variant: 'error',
-  title: 'Error: Danger Will Robinson',
-  children: singleParagraph(),
-  primaryActionLabel: 'Accept',
-  cancelLabel: 'Cancel',
-  secondaryActionLabel: 'Secondary button',
-  onPrimaryAction: action('primary'),
-  onSecondaryAction: action('secondary'),
-  onCancel: action('cancel'),
-  autoFocusButton: 'secondary'
-});
-
-AutoFocusSecondary.story = {
+export const AutoFocusSecondary = {
+  render: renderAlert,
+  args: {
+    ...Error.args,
+    secondaryActionLabel: 'Secondary button',
+    autoFocusButton: 'secondary'
+  },
   name: 'autoFocus secondary'
 };
 
-export const AutoFocusCancel = () => renderAlert({
-  variant: 'error',
-  title: 'Error: Danger Will Robinson',
-  children: singleParagraph(),
-  primaryActionLabel: 'Accept',
-  cancelLabel: 'Cancel',
-  secondaryActionLabel: 'Secondary button',
-  onPrimaryAction: action('primary'),
-  onSecondaryAction: action('secondary'),
-  onCancel: action('cancel'),
-  autoFocusButton: 'cancel'
-});
-
-AutoFocusCancel.story = {
+export const AutoFocusCancel = {
+  render: renderAlert,
+  args: {
+    ...Error.args,
+    secondaryActionLabel: 'Secondary button',
+    autoFocusButton: 'cancel'
+  },
   name: 'autoFocus cancel'
 };
 
 function renderAlert({...props}: SpectrumAlertDialogProps) {
   return (
     <div style={{display: 'flex', width: 'auto', margin: '100px 0'}}>
-      <DialogTrigger defaultOpen>
+      <DialogTrigger>
         <ActionButton>Trigger</ActionButton>
         <AlertDialog {...props} onPrimaryAction={action('primary')} onSecondaryAction={action('secondary')} onCancel={props.onCancel} />
       </DialogTrigger>
     </div>
   );
 }
+
+// TODO: should the button be immediately pending? Right now there is a delay
```

Comment:
Since a pending button only becomes disabled after 1 second, there is a period of time in which the user might try to trigger the primary/secondary button before it becomes pending. Probably not a problem if the button becomes pending in response to the press, but I can see a case where the AlertDialog's primary/secondary button should start in the pending state (e.g. maybe the dialog has something inside it that needs to load before allowing the user to confirm).

Note that pressing on the pending button before it gets its pending style doesn't actually do anything (intentional design) but could be visually confusing potentially?

Reply:
All of this is intentional and the delay is fairly short to counter that confusion. It prevents a flash to the pending state before very quickly moving to complete.
I think the other non-pending buttons should be disabled immediately though.

Reply:
maybe not visually disabled immediately though since the main one you clicked doesn't do that?

Reply:
makes sense, I'll see about blocking out the onPress on those buttons immediately and delaying setting isDisabled

Reply:
Hm I guess "Escape" should also be disabled then? Think this goes back to @ktabors's concern about if it could possibly trap users in the Dialog if the request is long/times out but the implementor doesn't clear out the pending button